### ISSUE TITLE
Small vertical separation problem in the sidebar using an unformatted list

### ISSUE THREAD

**Issue by PXR5** [saschaleib/dokuwiki-template-ad-hominem#50]
Hello, thank you for the nice ad-hominem template, I like it a lot. I am however experiencing a small layout problem with the sidebar. I tested originally with macOS Safari so I also validated using Firefox and it is the same. If I revert to using the default dokuwiki template the problem disappears.

The sidebar wikitext looks like this...
<img width="433" alt="Screenshot 2024-09-17 at 19 41 59" src="https://github.com/user-attachments/assets/19f6cfdc-6ce6-42c8-be66-2f236cb23fa6">

but the first block appears with a smaller vertical separation below its section header, I checked the source in the browser and the menu blocks look almost the same, just there is a blank line after the first para before the first break, which shouldn't matter but it's the only thing I could find that's different.

This is the sidebar as part of the start page...
<img width="429" alt="Screenshot 2024-09-17 at 19 01 48" src="https://github.com/user-attachments/assets/37ffa844-6c5a-4e9e-a9c1-ba273513db11">

When editing with the sidebar as the main window it appears as intended, the separation the same as the other blocks. The problem only appears when the sidebar page is actually in the sidebar...
<img width="500" alt="Screenshot 2024-09-17 at 19 06 31" src="https://github.com/user-attachments/assets/615d1a2c-dcb5-4ab0-8894-f1ac023c53c5">

Here is the in-browser source...
<img width="451" alt="Screenshot 2024-09-17 at 19 39 58" src="https://github.com/user-attachments/assets/9c619d01-95d3-4a06-aab1-4866bd41d97e">

The presentation looks as intended using the default dokuwiki template...
<img width="222" alt="Screenshot 2024-09-17 at 19 53 44" src="https://github.com/user-attachments/assets/6279e591-eff4-4b64-a85f-677280ddb5e1">

**Comment by saschaleib** [comment]
Hello Brian,

Thanks for you request. I had a look at your issue, but I have to say that because of the way that you structure the sidebar, I don't think I can (or rather: "should") help you there.

Let me first explain that one of the issues I had to work around when creating this template was that DokuWiki has a tendency to create unnecessary (empty) paragraphs and extra `<br>` line breaks, where they should not be. For this reason there are all kinds of special rules in the style sheets to hide or mitigate the styling issues that come from these extra elements.

However, they only work if the pages are done in a standards-conform way. Like, paragraphs and headlines, etc. are applied to the page content (including the sidebars) in a way that is consistent with the relevant guidelines – including (but not limited to) the [WCAG Accessibility guidelines](https://www.w3.org/TR/WCAG22/) – see for example, this article about [how to use headlines](https://www.w3.org/WAI/tutorials/page-structure/headings/).

What you are doing is that you _simulate_ headlines by using **bold** text (an absolute no-no) and then add line breaks to add spacing before and after. This is not compliant code, and should never be used in this way.

Instead, please try to use headlines instead (levels 4 and 5, or in DokuWiki code: `===` and `==` should work very well). These will automatically be styled in the headlines typeface (a serif font by default), and get appropriate spacing before and after. 

Alternatively, if you use a plugin to enable HTML tags (like my own [Ad-Hoc HTML](https://www.dokuwiki.org/plugin:adhoctags) Plugin), you can also use `<dl>`, `<dt>` and `<dd>` to structure the menu. These will get a bit “tighter” appearance, if this is wanted.

For an example, see this page here on my English wiki: [https://fallacies.online/wiki/logic/formal_fallacies/index](https://fallacies.online/wiki/logic/formal_fallacies/index). the same menu in the German version uses `<dl>`s instead: h[ttps://denkfehler.online/wiki/logik/fehlschluesse/hauptseite](https://denkfehler.online/wiki/logik/fehlschluesse/hauptseite) – this allows you to directly compare the different approaches.

I hope this helps. I leave the issue open for a while in case there are more questions, but I don't think that make any changes here, as this might actually encourage non-accessible bad practices in code, which I would rather not do.

Best greetings

/sascha

**Comment by PXR5** [comment]
Hello Sascha,

Thank you for your speedy reply. Yes, I can see from your explanation how I was using the wrong elements. I have replaced the  newlines and bolded type with headlines and the layout now looks very good, better than before and doesn't have any weird formatting issues now it's done properly. 

As you can probably ascertain this is my first personal wiki project and I have plenty of things to learn, the tip about using ad-hoc HTML will likely come in handy as will the references to the W3C docs and your dokuwiki sites.

Since there isn't any problem with the template I will just make this comment and leave it to you to decide how you want to close or even delete this issue.

Thanks a lot for your time and helpful support.

Best Regards,
Brian

**Comment by saschaleib** [comment]
Hi again, no worries, we have all started sometime and trying and asking is the best way to learn :-)

Just because Accessibility is very close to my heart, allow me to add some more explanations why this particular structure is so important: not everybody is using a visual web browser - e.g. people with visual disabilities (read: blind people) use what is known as a screen reader, which reads the content of the page instead of showing it.

Now, it is a lot easier to get an overview of a page if you can see it rather than when it is read to you - and so these readers have a function to first only read the headlines, and if you are interested in more detail, then they can go down to "smaller" headlines, until the thing you are looking for is found - and only then the actual text. 

This of course only works if the headlines can be detected, i.e. if they are marked up as actual headlines - this will not work if they are just simple paragraphs with a bold style – even though _visually_ those still _look like_ headlines...

So in short, it is not (only) about looking good, but also about providing a small but very helpful little assistance to people who need it.

And, frankly, I don't think it is a very big sacrifice to make. Those headlines even actually look better as well :-)

In any case, I'll close it here now. Have a good time and enjoy your wiki :-)

Best greetings

/sascha
